src/Data/List/Membership/Setoid/Properties.agda

```diff
@@ -0,0 +1,206 @@
+-- concat
+
+module _ {c ℓ} (S : Setoid c ℓ) where
+
+  open Setoid S using (_≈_)
+  open Membership S using (_∈_)
+  open Equality S using (≋-setoid)
+  open Membership ≋-setoid using (find) renaming (_∈_ to _∈≋_)
+
+  ∈-concat⁺ : ∀ {v xss} → Any (v ∈_) xss → v ∈ concat xss
+  ∈-concat⁺ = Any.concat⁺
+
+  ∈-concat⁻ : ∀ {v} xss → v ∈ concat xss → Any (v ∈_) xss
+  ∈-concat⁻ = Any.concat⁻
+
+  ∈-concat⁺′ : ∀ {v vs xss} → v ∈ vs → vs ∈≋ xss → v ∈ concat xss
+  ∈-concat⁺′ v∈vs = ∈-concat⁺ ∘ Any.map (flip (∈-resp-≋ S) v∈vs)
+
+  ∈-concat⁻′ : ∀ {v} xss → v ∈ concat xss → ∃ λ xs → v ∈ xs × xs ∈≋ xss
+  ∈-concat⁻′ xss v∈c[xss] with find (∈-concat⁻ xss v∈c[xss])
+  ... | xs , t , s = xs , s , t
+
+------------------------------------------------------------------------
+-- applyUpTo
+
+module _ {c ℓ} (S : Setoid c ℓ) where
+
+  open Setoid S using (_≈_; refl)
+  open Membership S using (_∈_)
+
+  ∈-applyUpTo⁺ : ∀ f {i n} → i < n → f i ∈ applyUpTo f n
+  ∈-applyUpTo⁺ f = Any.applyUpTo⁺ f refl
+
+  ∈-applyUpTo⁻ : ∀ {v} f {n} → v ∈ applyUpTo f n →
+                 ∃ λ i → i < n × v ≈ f i
+  ∈-applyUpTo⁻ = Any.applyUpTo⁻
+
+------------------------------------------------------------------------
+-- tabulate
+
+module _ {c ℓ} (S : Setoid c ℓ) where
+
+  open Setoid S using (_≈_; refl) renaming (Carrier to A)
+  open Membership S using (_∈_)
+
+  ∈-tabulate⁺ : ∀ {n} {f : Fin n → A} i → f i ∈ tabulate f
+  ∈-tabulate⁺ i = Any.tabulate⁺ i refl
+
+  ∈-tabulate⁻ : ∀ {n} {f : Fin n → A} {v} →
+                v ∈ tabulate f → ∃ λ i → v ≈ f i
+  ∈-tabulate⁻ = Any.tabulate⁻
+
+------------------------------------------------------------------------
+-- filter
+
+module _ {c ℓ p} (S : Setoid c ℓ) {P : Pred (Carrier S) p} where
+
+  open Setoid S using (_≈_; sym)
+  open Membership S using (_∈_)
+
+  ∈-filter⁺ : (P? : Decidable P) → P Respects _≈_ →
+              ∀ {v xs} → v ∈ xs → P v → v ∈ filter P? xs
+  ∈-filter⁺ P? resp {v} {x ∷ _} (here v≈x)   Pv with P? x
+  ... | yes _   = here v≈x
+  ... | no  ¬Px = contradiction (resp v≈x Pv) ¬Px
+  ∈-filter⁺ P? resp {v} {x ∷ _} (there v∈xs) Pv with P? x
+  ... | yes _ = there (∈-filter⁺ P? resp v∈xs Pv)
+  ... | no  _ = ∈-filter⁺ P? resp v∈xs Pv
+
+  ∈-filter⁻ : (P? : Decidable P) → P Respects _≈_ →
+              ∀ {v xs} → v ∈ filter P? xs → v ∈ xs × P v
+  ∈-filter⁻ P? resp {v} {[]}     ()
+  ∈-filter⁻ P? resp {v} {x ∷ xs} v∈f[x∷xs] with P? x
+  ... | no  _  = Prod.map there id (∈-filter⁻ P? resp v∈f[x∷xs])
+  ... | yes Px with v∈f[x∷xs]
+  ...   | here  v≈x     = here v≈x , resp (sym v≈x) Px
+  ...   | there v∈fxs = Prod.map there id (∈-filter⁻ P? resp v∈fxs)
+
+------------------------------------------------------------------------
+-- length
+
+module _ {c ℓ} (S : Setoid c ℓ) where
+
+  open Membership S using (_∈_)
+
+  ∈-length : ∀ {x xs} → x ∈ xs → 1 ≤ length xs
+  ∈-length {_} {_ ∷ xs} (here px)    = s≤s z≤n
+  ∈-length {_} {_ ∷ xs} (there x∈xs) =
+    ≤-trans (∈-length x∈xs) (n≤1+n (length xs))
+
```

Comment:
Can't we have `n≤1+n (length _)` and drop the pattern matching on implicit arguments?

Reply:
Nope but we can have `n≤1+n _`!